src/algebra/ordered_ring.lean

```diff
@@ -121,31 +130,48 @@ lemma le_mul_of_one_le_left (hb : 0 ≤ b) (h : 1 ≤ a) : b ≤ a * b :=
 suffices 1 * b ≤ a * b, by rwa one_mul at this,
 mul_le_mul_of_nonneg_right h hb
 
+section
+variable [nontrivial α]
+
 lemma bit1_pos (h : 0 ≤ a) : 0 < bit1 a :=
 lt_add_of_le_of_pos (add_nonneg h h) zero_lt_one
 
-lemma bit1_pos' (h : 0 < a) : 0 < bit1 a :=
-bit1_pos h.le
-
 lemma lt_add_one (a : α) : a < a + 1 :=
 lt_add_of_le_of_pos le_rfl zero_lt_one
 
 lemma lt_one_add (a : α) : a < 1 + a :=
 by { rw [add_comm], apply lt_add_one }
 
+end
+
+lemma bit1_pos' (h : 0 < a) : 0 < bit1 a :=
+begin
+  letI := nontrivial_of_lt _ _ h,
```

Comment:
Why not `nontriviality`?

Reply:
Great. I changed this one, and 3 others.